aten/src/ATen/core/ivalue_inl.h

```diff
@@ -1175,9 +1175,9 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
   // set up before running the callback, as in, it will set up the CUDA streams,
   // synchronize them with the value, and so on (if needed).
   template<typename T>
-  void invokeCallback(T callback, bool uses_future) {
+  void invokeCallback(const T& callback, bool uses_future) {
```

Comment:
I think this actually might break SFINAE unfortunately.

Reply:
Use T&